src/components/DeploymentFormV2.vue

```diff
@@ -96,6 +105,7 @@
   const infrastructureOverrides = ref(stringify(props.deployment.infrastructureOverrides))
   const enforceParameterSchema = ref(props.deployment.enforceParameterSchema)
   const shouldValidateParameters = ref(true)
+  const { kind } = usePrefectKind(parameters)
```

Comment:
not new in this PR but huh i was surprised this is intended to be mutated from outside 🤨

Reply:
What do you mean? The `kind` or the `parameters`?

Reply:
the `kind`. was just surprised seeing that get mutated directly up in the template.

Reply:
gotcha, yeah we don't do it often but its not an uncommon pattern to have a composition return a ref that you can mutate and then based on that mutation the composition does something. In this case its tracking some in memory values for parameters so that when you switch between kinds it remembers what you had before.